### PR TITLE
Updated Readme 2.1 

### PR DESCRIPTION
**Pull request by bhagyashree-sarawate**
@raunakkumar ,Shanbhag, PAVAN Subbayya please review 
Added documentation for helm chart upgradation from 2.0.0 to 2.1.0

### CODE REVIEW

File: helm/charts/hpe-csi-driver/README.md
Comment (raunakkumar) [inline helm/charts/hpe-csi-driver/README.md:73]: We can still keep this section and add an exception for upgrade to 2.1.0?

Comment (datamattsson) [reply helm/charts/hpe-csi-driver/README.md:73]: Yes, keep this.

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [reply helm/charts/hpe-csi-driver/README.md:73]: Add a `## Upgrading 2.0.0 to 2.1.0` heading.

Comment (datamattsson) [reply helm/charts/hpe-csi-driver/README.md:73]: Please use spell check when submitting PRs.

Rewrite L73 as follows:
Before version 2.0.0 is uninstalled, the following CRDs needs to be updated.

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [reply helm/charts/hpe-csi-driver/README.md:80]: Rewrite L80: 
If there are HPE Alletra 9000, Primera or 3PAR Remote Copy Groups configured on the cluster, submit the following `Job`:

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:82]: Delete.

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:83]: This needs to be rewritten as a Job controller: https://kubernetes.io/docs/concepts/workloads/controllers/job/

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:85]: Wait for the the `Job` to complete:
```
kubectl wait --for=condition=complete --timeout=600s job/convert-rcg-info
```

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:88]: Next, apply the new CRDs.
```
kubectl apply -f https://raw.githubusercontent.com/hpe-storage/co-deployments/master/helm/charts/hpe-csi-driver/crds/hpevolumeinfos_v2_crd.yaml
kubectl apply -f https://raw.githubusercontent.com/hpe-storage/co-deployments/master/helm/charts/hpe-csi-driver/crds/hpevolumegroupinfos_v2_crd.yaml
kubectl apply -f https://raw.githubusercontent.com/hpe-storage/co-deployments/master/helm/charts/hpe-csi-driver/crds/snapshotgroupinfos_v2_crd.yaml
kubectl apply -f https://raw.githubusercontent.com/hpe-storage/co-deployments/master/helm/charts/hpe-csi-driver/crds/hpereplicated_deviceinfo_v2_crd.yaml
```

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:86]: Not needed.

**Review by datamattsson — CHANGES_REQUESTED** [review]

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [reply helm/charts/hpe-csi-driver/README.md:83]: We also need to remove the infinity sleep from the rcg-info binary and rebuild the image.

Comment (datamattsson) [reply helm/charts/hpe-csi-driver/README.md:83]: Moved manifest to its own release directory: https://raw.githubusercontent.com/hpe-storage/co-deployments/master/yaml/rcg-info/v1.0.0/convert-rcg-info.yaml

File: helm/charts/hpe-csi-driver/README.md
Comment (raunakkumar) [reply helm/charts/hpe-csi-driver/README.md:87]: @bhagyashree-sarawate are you planning to update convert-rcg-info into a job?

Comment (bhagyashree-sarawate) [reply helm/charts/hpe-csi-driver/README.md]: Done the job changes,

File: helm/charts/hpe-csi-driver/README.md
Comment (bhagyashree-sarawate) [reply helm/charts/hpe-csi-driver/README.md]: > We also need to remove the infinity sleep from the rcg-info binary and rebuild the image.

Addressed  this in PR 
https://github.hpe.com/hpe/hpe_3par_primera_rcginfo/pull/2

File: yaml/rcg-info/v1.0.0/convert-rcg-info.yaml
Comment (datamattsson) [inline yaml/rcg-info/v1.0.0/convert-rcg-info.yaml:5]: Omit namespace.

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:83]: Only encase commands code blocks, not descriptive comments.

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:87]: Rewrite this into: 
Change `kubectl` context into the `Namespace` where the HPE CSI Driver is installed. The most common is "hpe-storage".

```
kubectl config set-context --current --namespace=hpe-storage
```

Create the `Job` using the below commands, which will modify the "rcg-info" record to the new key "RCGCreatedByCSP".

```
kubectl apply -f https://raw.githubusercontent.com/hpe-storage/co-deployments/master/yaml/rcg-info/v1.0.0/convert-rcg-info.yaml
```

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:89]: "using the below command."

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:91]: Omit the output and remove "-n hpe-storage"

File: helm/charts/hpe-csi-driver/README.md
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:82]: Add punctuation to "uninstallation."

**Review by datamattsson — CHANGES_REQUESTED** [review]
A few clarifications.

**Review by datamattsson — CHANGES_REQUESTED** [review]
Submitted a PR [here](https://github.com/bhagyashree-sarawate/co-deployments/pull/3) to ensure users won't get ahead of themselves.

**Comment by bhagyashree-sarawate** [comment]
> Submitted a PR [here](https://github.com/bhagyashree-sarawate/co-deployments/pull/3) to ensure users won't get ahead of themselves.

Should I close this PR ?

**Comment by datamattsson** [comment]
> > Submitted a PR [here](https://github.com/bhagyashree-sarawate/co-deployments/pull/3) to ensure users won't get ahead of themselves.
> 
> Should I close this PR ?

If you merge my PR into your private branch you can push the commit into this PR.

**Comment by bhagyashree-sarawate** [comment]
> > > Submitted a PR [here](https://github.com/bhagyashree-sarawate/co-deployments/pull/3) to ensure users won't get ahead of themselves.
> > 
> > 
> > Should I close this PR ?
> 
> If you merge my PR into your private branch you can push the commit into this PR.

Done

**Review by datamattsson — APPROVED** [review]
LGTM

**Comment by bhagyashree-sarawate** [comment]
@Raunak , @datamattsson Please do not merge this PR , till we get confirmation from test team and Pradip

**Comment by bhagyashree-sarawate** [comment]
> 



> 

We can merge this PR if looks good
Got confirmation from test team , we can go ahead with this change

**Pull request merged**
